pokemongo_bot/cell_workers/egg_incubation_worker.py

```diff
@@ -0,0 +1,130 @@
+        if len(self.available_incubators) and len(self.ready_eggs):
+            logger.log('[#] Applying incubators to eggs...')
+            self._incubate_eggs()
+            self._calculate_remaining_km() # get updated min km target
+        if len(self.used_incubators) and (self.km_walked!=EggIncubationWorker.last_km_walked or check_incubation):
+            logger.log('[#] Next egg hatches in {:.2n} km.'.format(min([km if km>=0 else 0 for km in self.remaining_km])),'yellow')
+            EggIncubationWorker.last_km_walked=self.km_walked
+
+    def _calculate_remaining_km(self):
+        self.remaining_km = [(lambda x: x.get("target_km_walked",10000)-self.km_walked)(x) for x in self.used_incubators]
+
+    def _check_hatched_eggs(self):
+        hatch_success = False
+        self.bot.api.get_hatched_eggs()
+        response_dict = self.bot.api.call()
+        logger.log('[d] Get hatched eggs call result: {}'.format(response_dict))
+        try:
+            hatched_results = response_dict['responses']['GET_HATCHED_EGGS']
+            if 'pokemon_id' in hatched_results:
+                hatch_success = True
+                pokemon = self.bot.pokemon_list[hatched_results['pokemon_id']]['Name']
+                xp = hatched_results['experience_awarded']
+                candy = hatched_results['candy_awarded']
+                ziggy = hatched_results['stardust_awarded']
+                logger.log('[#] Congratulations! You\'re the proud trainer of a new {}!'.format(pokemon),'green')
+                logger.log('[#] Acquired {} xp, {} candies, and {} stardust from hatch.'.format(xp,candy,ziggy),'green')
+            elif 'Protobuf' in hatched_results:
+                logger.log('[#] Something probably hatched, but we couldn\'t tell what. Update your protos!','yellow')    
+                hatch_success = True
+            else:
+                logger.log('[#] No hatched eggs this time')
+        except Exception as e:
+            hatch_success = True 
+            logger.log('[#] Hatched eggs check call failed!','red')
+        return hatch_success
+
+    def _check_inventory(self):
+        response_dict = self.bot.get_inventory()
+        incubator_list = []
+        egg_list = []
+        try:
+            for item in response_dict['responses']['GET_INVENTORY']['inventory_delta']['inventory_items']:
+                if 'inventory_item_data' in item:
+                    item = item['inventory_item_data']
+                    if 'egg_incubators' in item:
+                        incubators = item['egg_incubators']
+                        if isinstance(incubators, list):
+                            #new protos
+                            for inc in incubators:
+                                incubator_list.append(inc['egg_incubator'])
+                        else:
+                            #old protos
+                            incubator_list.append(incubators['egg_incubator'])
```

Comment:
This line not work. Change to:
`for inc in incubators['egg_incubator']:
      incubator_list.append(inc)
`